### DISCUSSION TITLE
Visit stops in fixed sequence

### DISCUSSION THREAD

**Discussion by TeodoraBogdan** [google/or-tools#3781] [Routing (and legacy CP) questions]
Hello!
 
I am trying to force some stops to be visited in an exact order. The stops have time-window and capacity. And the vehicles have start time, end time and max capacity.
I create a new dimension to force the stops visitation order.

I have:

```
fixedEvaluatorIndices = m_routing->RegisterTransitCallback(
			[this](int64 from_index, int64 to_index) -> int64 {
				auto from_node = manager->IndexToNode(from_index);
				auto to_node = manager->IndexToNode(to_index);

				if(from_node and to_node are in the fixedSeq)
					return 0;
				return 1000;
			});
m_routing->AddDimensionWithVehicleCapacity(fixedEvaluatorIndices, 0, std::vector<int64>(m_vehiclesSize, INT64_MAX), true, "FixedSeq");

operations_research::RoutingDimension* dimension = m_routing->GetMutableDimension("FixedSeq");
operations_research::RoutingDimension* Tdimension = m_routing->GetMutableDimension("Time");

```

I go through the fixed sequence list and set the stops to be visited by the same vehicle, the fixed sequence dimension value to be equal and the time dimension value to be increasing.

```
solver.AddConstraint(solver.MakeEquality(m_routing->VehicleVar(prevStopIndex), m_routing->VehicleVar(nodeIndex)));

solver.AddConstraint(solver.MakeEquality(dimension->CumulVar(prevStopIndex), dimension->CumulVar(nodeIndex)));
solver.AddConstraint(solver.MakeLessOrEqual(Tdimension->CumulVar(prevStopIndex), Tdimension->CumulVar(nodeIndex)));
```

If the stops from the fixed sequence have a time-window set, no result is found. The time-window is very generous. And the time dimension has a big slack value.

I didn't want to put the time dimension to return 0 for the fixed sequence stops, because it is a possibility to mess the the time-window

What do I have to set to make it work?

**Comment by lperron** [comment]
Why not merge the first to the last node in the sequence into one node ?
